### PR TITLE
Get-DbaSsisExecutionHistory update

### PR DESCRIPTION
**Pull request by potatoqualitee**
will start a review to show what i changed and why, one moment

### CODE REVIEW

File: functions/Get-DbaSsisExecutionHistory.ps1 [hunk 1]
```diff
@@ -49,12 +49,12 @@ function Get-DbaSsisExecutionHistory {
 
             Shows what would happen if the command were executed and would return the SQL statement that would be executed per instance.
     #>
-    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
```
Comment (potatoqualitee) [inline functions/Get-DbaSsisExecutionHistory.ps1]: parametersets are only required when you have conflicting params, like..

"If you run Copy-DbaDatabase, you can run EITHER backuprestore OR detachattach"  or "database is ONLY mandatory when..."

SupportsShoudlProcess is for whatif and is only used when there's modification to settings/data/etc.

File: functions/Get-DbaSsisExecutionHistory.ps1 [hunk 2]
```diff
@@ -49,12 +49,12 @@ function Get-DbaSsisExecutionHistory {
...
+    [CmdletBinding()]
     param (
         [parameter(Mandatory = $true)]
-        [DbaInstanceParameter]$SQLInstance,
+        [DbaInstanceParameter]$SqlInstance,
         [PSCredential]$SqlCredential,
-        [ValidateSet("Created", "Running", "Cancelled", "Failed", "Pending", "Halted", "Succeeded", "Stopping","Completed")]
+        [ValidateSet("Created", "Running", "Cancelled", "Failed", "Pending", "Halted", "Succeeded", "Stopping", "Completed")]
```
Comment (potatoqualitee) [inline functions/Get-DbaSsisExecutionHistory.ps1]: My editor did that ;)

File: functions/Get-DbaSsisExecutionHistory.ps1 [hunk 3]
```diff
@@ -49,12 +49,12 @@ function Get-DbaSsisExecutionHistory {
 
             Shows what would happen if the command were executed and would return the SQL statement that would be executed per instance.
     #>
-    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+    [CmdletBinding()]
     param (
         [parameter(Mandatory = $true)]
-        [DbaInstanceParameter]$SQLInstance,
+        [DbaInstanceParameter]$SqlInstance,
```
Comment (potatoqualitee) [inline functions/Get-DbaSsisExecutionHistory.ps1]: our standard, like Microsoft's is to use Sql and not SQL within PS vars

File: functions/Get-DbaSsisExecutionHistory.ps1 (hunk 2 above)
Comment (ChrisTuckerNM) [reply functions/Get-DbaSsisExecutionHistory.ps1]: Cute

File: functions/Get-DbaSsisExecutionHistory.ps1 [hunk 4]
```diff
@@ -65,50 +65,54 @@ function Get-DbaSsisExecutionHistory {
...
-
-            if ($Environment) {
-                $csv = "`'" + ($Environment -join"'","'") + "`'"
-                $environmentq = "AND e.[environment] in ($csv)"
-            } else {
-                $environmentq = ''
-            }
-
-            $q = @"
+            'Created'   = 1
```
Comment (potatoqualitee) [inline functions/Get-DbaSsisExecutionHistory.ps1]: caps cuz that's what we usually see from PowerShell results.

File: functions/Get-DbaSsisExecutionHistory.ps1 [hunk 5]
```diff
@@ -65,50 +65,54 @@ function Get-DbaSsisExecutionHistory {
...
+        else {
+            $environmentq = ''
+        }
+        
+        $sql = "
             WITH
             cteLoglevel as (
                 SELECT
-                    execution_id,
-                    cast(parameter_value AS INT) AS logging_level
```
Comment (potatoqualitee) [inline functions/Get-DbaSsisExecutionHistory.ps1]: logging_level is SQLy, while LoggingLevel is PowerShelly so I changed the output to look more like PS and less like SQL

**Comment by potatoqualitee** [comment]
@ChrisTuckerNM - can you test this against a server that has some executions on it, making sure to include various params like status, etc? ty

**Comment by niphlod** [comment]
good now, except for no parametrized queries. Lemme know when I can chime in, so everyone can see how easy it is

**Comment by ChrisTuckerNM** [comment]
I tested all parameters except for credentials and multiple instances... Is there an issue?

**Comment by niphlod** [comment]
yeah, even if theoretically a user dabbling with dbatools is a "sane" user, constructing the query via string interpolation is susceptible to SQL Injection......

**Comment by ChrisTuckerNM** [comment]
Oops, @potatoqualitee IS there an issue? I tested with all parameters against an instance with tons of executions, including failures and cancels.

**Comment by ChrisTuckerNM** [comment]
@niphlod yeah, that last comment was meant for CL, not you, I agree on the injection, I was gonna have a crack at it but fire away..

**Comment by niphlod** [comment]
I use this snippet, it's easily sprinkable throughout this module. If you have questions about implementing it ask away

```
function Get-SqlOrIfied([object[]]$element, [string]$keyprefix) {
            $stmt = ''
            $kv = [ordered]@{}
            for ($i=0; $i -lt $element.length; $i++) {
	            $kv["@$($keyprefix)_$($i)"] = $element[$i]
            }
            $stmt = '(' + ($kv.Keys -Join ', ') + ')'
            [pscustomobject]@{
                stmt = $stmt
                params = $kv
            }
        }
        ......
        $query_params = @{}

        if ($Status) {
            $piece = Get-SqlOrIfied -element $statuses[$Status] -keyprefix 'status'
            $query_params += $piece.params 
            $statusq = "AND e.[Status] IN $($piece.stmt)"
        }
        else {
            $statusq = ''
        }
```

**Comment by niphlod** [comment]
Note for later, we definitely need a -Since parameter (and maybe with a week old default that trims at a sane level the retrieved history)

**Comment by potatoqualitee** [comment]
this is breakin all my tests so imma merge

**Comment by wsmelton** [comment]
@niphlod unless you have been going through updating commands the method used in this command for dynamic T-SQL is how it is done in every command we have...and I mean every one literally.

**Comment by niphlod** [comment]
@wsmelton : and that's exactly why we should change. BTW, it's not "safe by default" but others supposedly "interpolate" parameters that come from autocomplete, and things such databasename . Here there are several parameters "in free text" so it's more "urgent"

**Pull request merged**